test/e2e/e2e_test.go

```diff
@@ -28,8 +28,65 @@ import (
 
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ecv1alpha1 "go.etcd.io/etcd-operator/api/v1alpha1"
 )
 
+// TestZeroMemberCluster tests if zero member Etcd Cluster does not create its StatefulSet
+func TestZeroMemberCluster(t *testing.T) {
+	feature := features.New("zero-member-cluster")
+	etcdClusterName := "etcd-cluster-zero"
+	size := 0
+
+	etcdClusterSpec := &ecv1alpha1.EtcdCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "operator.etcd.io/v1alpha1",
+			Kind:       "EtcdCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      etcdClusterName,
+			Namespace: namespace,
+		},
+		Spec: ecv1alpha1.EtcdClusterSpec{
+			Size:    size,
+			Version: "v3.5.18",
```

Comment:
nit: Can we use the `ETCD_VERSION` variable here? 
Ref: https://github.com/etcd-io/etcd-operator/blob/main/Makefile#L77